### PR TITLE
ELEMENTS-1771:Allow option to configure Integer display Formatting nuxeo-checkbox-aggregation.js

### PR DESCRIPTION
**Pull request by yashgupta-hyland**
https://hyland.atlassian.net/browse/ELEMENTS-1771
Backward port

### CODE REVIEW

File: ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js [hunk 1]
```diff
@@ -257,6 +255,19 @@ import { AggregationBehavior } from './nuxeo-aggregation-behavior.js';
       this.setAttribute('tabindex', 0);
     }
 
+    _formatDocCount(docCount) {
+      // Fetch the property value from web-ui-properties.xml
+      const isNumberFormattingEnabled =
+        Nuxeo && Nuxeo.UI && Nuxeo.UI.config && Nuxeo.UI.config.numberFormattingEnabled !== undefined
```
Comment (madhurkulshrestha-hyland) [inline ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: no need to check for "!== undefined". remove this part

File: ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js [hunk 2]
```diff
@@ -257,6 +255,19 @@ import { AggregationBehavior } from './nuxeo-aggregation-behavior.js';
...
+          ? Nuxeo.UI.config.numberFormattingEnabled
+          : false; // Default to false if the property is not set
+
+      if (isNumberFormattingEnabled) {
+        return new Intl.NumberFormat().format(docCount); // Apply formatting if enabled
+      }
+      return docCount.toString(); // Return if formatting is disabled
```
Comment (madhurkulshrestha-hyland) [inline ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: is toString conversion required ? The previous code doesnt use this conversion

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js (hunk 2 above)
Comment (yashgupta-hyland) [reply ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: In most cases, this avoids unexpected type issues. and won’t impact performance, even if applied repeatedly or on large datasets. It’s a safe addition for consistency without any noticeable drawbacks.

Comment (madhurkulshrestha-hyland) [reply ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: if docCount is null, this will throw exception

**Review by madhurkulshrestha-hyland — CHANGES_REQUESTED** [review]

File: ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js (hunk 1 above)
Comment (yashgupta-hyland) [reply ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: removed

File: ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js (hunk 2 above)
Comment (yashgupta-hyland) [reply ui/nuxeo-aggregation/nuxeo-checkbox-aggregation.js]: Got the point , removed toString()

**Review by madhurkulshrestha-hyland — APPROVED** [review]

**Review by rahuljain-dev — APPROVED** [review]

**Review by alokhyland — APPROVED** [review]

**Pull request merged**